classes/core/PKPSessionGuard.php

```diff
@@ -0,0 +1,273 @@
+<?php
+
+namespace PKP\core;
+
+use Carbon\Carbon;
+use PKP\user\User;
+use APP\facades\Repo;
+use PKP\core\Registry;
+use APP\core\Application;
+use PKP\security\Validation;
+use InvalidArgumentException;
+use Illuminate\Auth\SessionGuard;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Date;
+use Illuminate\Contracts\Session\Session;
+use Symfony\Component\HttpFoundation\Cookie;
+use Illuminate\Contracts\Auth\Authenticatable as AuthenticatableContract;
+
+class PKPSessionGuard extends SessionGuard
+{
+    /**
+     * The currently authenticated user.
+     *
+     * @var \Illuminate\Contracts\Auth\Authenticatable|\PKP\user\User|null
+     */
+    protected $user;
+
+    /**
+     * The user provider implementation.
+     *
+     * @var \Illuminate\Contracts\Auth\UserProvider|\PKP\core\PKPUserProvider
+     */
+    protected $provider;
+
+    /**
+     * update the current user without firing any events or changes
+     *
+     * @param  \Illuminate\Contracts\Auth\Authenticatable|User  $user
+     * @return $this
+     */
+    public function updateUser(\Illuminate\Contracts\Auth\Authenticatable|\PKP\user\User $user)
```

Comment:
thats my plan but in few cases we need to add those otherwise `php intelephense` is marking those as error which are not as we have some of our own implementation but I tried to reduce those as much as possible .